src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/WebRequestSession.cs

```diff
@@ -74,39 +111,92 @@ public class WebRequestSession
         /// </summary>
         public int RetryIntervalInSeconds { get; set; }
 
+        /// <summary>
+        /// Gets or sets the SslProtocol for the session.
+        /// </summary>
+        public WebSslProtocol SslProtocol { get => _sslProtocol; set => SetStructVar(ref _sslProtocol, value); }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="WebRequestSession"/> class.
         /// </summary>
         public WebRequestSession()
         {
-            // build the headers collection
+            _changed = true;
+
+            // Build the headers collection
             Headers = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
             ContentHeaders = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
-            // build the cookie jar
-            Cookies = new CookieContainer();
+            // Build the cookie jar
+            _cookies = new CookieContainer();
 
-            // initialize the credential and certificate caches
-            UseDefaultCredentials = false;
-            Credentials = null;
-            Certificates = null;
+            // Initialize the credential and certificate caches
+            _useDefaultCredentials = false;
+            _credentials = null;
+            _certificates = null;
 
-            // setup the default UserAgent
+            // Setup the default UserAgent
             UserAgent = PSUserAgent.UserAgent;
 
-            Proxy = null;
-            MaximumRedirection = -1;
+            _proxy = null;
+            _maximumRedirection = -1;
+            _allowAutoRedirect = true;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether a new handler must be created.
+        /// </summary>
+        internal bool Changed => _changed;
```

Comment:
If follow my previous comment we have no need this since if we try to change a property already started HttpClient/handler we can catch InvalidOperationException
and set _handler to null. (Of cause we need GetHttpClient/GetHttpClientHandler method in WebSession.)

Reply:
See my other comments for the limitations of the approach you propose. It is probably a simpler approach but only if we don't care that any use of httpclienthandler affecting parameters on the command line with a WebSession are going to cause non-persistence.

Reply:
We can use only `Gets` and check if an argument was changed and if so we recreate the target object.